distributed/recreate_exceptions.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import print_function, division, absolute_import
+
+import logging
+from tornado import gen
+from .client import futures_of
+from.worker import _deserialize
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionScheduler(object):
+    """ A plugin for the scheduler to recreate exceptions locally
+
+    This adds the following routes to the scheduler
+
+    *  channel-subscribe
+    *  channel-unsubsribe
+    *  channel-append
+    """
+    def __init__(self, scheduler):
+        self.scheduler = scheduler
+
+        handlers = {'cause_of_failure': self.cause_of_failure}
+
+        self.scheduler.handlers.update(handlers)
```

Comment:
Perhaps combine the two lines above

    self.scheduler.handlers['cause_of_failure'] = self.cause_of_failure